### ISSUE TITLE
Understanding training losses and CenterNet model script

### ISSUE THREAD

**Issue by YashRunwal** [yjh0410/CenterNet-plus#10]
Hi,

Using grayscale images of size (512, 1536), if I try to evaluate the model using the train.py script wherein the argument trainable=False for the model then we get the following shapes for the variables:
```
topk_bbox_pred: (100,4)
topk_scores: (100,)
topk_cls_inds (100,)
```

It is clear that the number 100 indicates the top 100 predictions which are defined in the class argument `topk=100`.  
What do these 100 predictions indicate though? Are these per pixel predictions? i.e. top pixels having the max values?
But how do we evaluate these predictions? i.e. how do we plot these predictions on the images? 

Similarly for training mode:
The shape of `p2 : [batch, 128, 384, channels]` and therefore the shape is [batch, 128*384, channels]. In `create_gt `script, `gt_creator` function actually creates a tensor of `[128*384]` size in this case. Does that mean it for every pixel?, i.e. loss is calculated for each pixel?

Please clear these doubts if possible.

Thank You.

**Comment by yjh0410** [comment]
You can use the test.py code file to plot these predictions on the images.
I creator targets for every pixels in P2 feature. For heatmap loss, it computes all the pixels' loss, but txty loss, twth loss and iou loss are computed on the positive samples(the pixels correspond the center points).

**Comment by YashRunwal** [comment]
Okay! So that's why it has the shape (128*284) in my case. I think I understood that now. @yjh0410 Can you please take a look here (https://github.com/yjh0410/CenterNet-plus/issues/9#issue-947032433) ? I have a doubt here in the gt_creator which is explained in the question. I think this is important as there will be a lot of scenarios wherein we cannot resize the image.

**Comment by YashRunwal** [comment]
@yjh0410 Hi, 

So I have changed the code to use [1, 128, 384, channels] but now am stuck at the following line of code:

`bbox_pred = torch.clamp((self.decode_boxes(txtytwth_pred) / self.input_size)[0], 0., 1.)`

The output of decode_boxes is for me of the shape `[1, 128, 384, 4]`
Here above, you divide the decode_boxes(txtytwth_pred) by input_size. However, my input_size is (512, 1536) so with which value should I divide?
